### ISSUE TITLE
wait for SRE to be ready

### ISSUE THREAD

**Issue by pkra** [pkra/mathjax-node-sre#22]
We don't handle SRE's odd asynchronicity for `SRE.ready`.

Until https://github.com/zorkow/speech-rule-engine/issues/247 is fixed, we need a timeout/restart hack.

**Comment by pkra** [comment]
here's one example
 
```js
let restart = function() {
  if (!sre.engineReady()) {
    setTimeout(restart, 200);
    return;
  }
  main()
};
```

**Comment by pkra** [comment]
Thanks to the amazing @zorkow, it seems `mode: "sync"` is sufficient, cf. https://github.com/zorkow/speech-rule-engine/issues/247#issuecomment-405988363

**Comment by pkra** [comment]
Hm... Seems like `mode: "sync"` is not sufficient after all.

The following will fail the first test -- but if you switch the order, it's fine (presumably because MathJax takes a longer time).

```js
var tape = require('tape');
const mj = require('mathjax-node').typeset;
const postprocessor = require('../lib/main.js').postprocessor;
const preprocessor = require('../lib/main.js').preprocessor;

tape('Wait for sre.engineReady()', function(t) {
  t.plan(2);

  const input1 = {math: '(', format: "TeX", mmlNode: true};
  preprocessor(input1, function(output){
    console.log(output.math)
    t.ok(output.math.includes('left-parenthesis'), 'Preprocessor enrichment ok');
  });

  const input2 = {math: '(', format: "TeX", mml: true, mmlNode: true};

  mj(input, function(result, input){
    postprocessor({speakText: true}, result, input2, function(result){
      t.equal(result.speakText, 'left-parenthesis', 'Postprocessor ok');
    })
  });
});
```

**Comment by pkra** [comment]
Thank you @zorkow for pushing out a fix in SRE v2.2.3-beta.0. I'll prep a proper patch.
